lua/lspconfig/ui/lspinfo.lua

```diff
@@ -144,16 +145,61 @@ local function make_client_info(client)
   return lines
 end
 
+--- if the server lists length longer than window it should be show in a
+--- new line
+--- | LspInfo window                                       | --out of window
+--- | Configured servers list:  server_name, server_name1, | server_name2
+--- |                           server_name3, server_name4,| --new change
+---@private
+local function generate_servers_list(servers_ctx)
+  local header, servers, width = servers_ctx.header, servers_ctx.servers, servers_ctx.width
+  local lines, hi_scope = {}, {}
+  local space = ' '
+  local server_name_sep = ',' .. space
+  local length, start_index, _end_index = #header, 1, 0
+
+  for i = 1, #servers do
+    local next_index = i == #servers and i or i + 1
+    -- check the lines is empty or not
+    local header_empty = next(lines) == nil and true or false
+    length = length + #server_name_sep + #servers[i]
+    table.insert(hi_scope, { length - #server_name_sep - #servers[i], length - #server_name_sep })
+
+    if (length + #servers[next_index]) > width or length == width then
+      _end_index = i
+    end
+
+    -- when the index is last one and still has the server names not insert
+    if i == #servers and length ~= 0 then
+      _end_index = i
+    end
+
+    if _end_index > 0 then
+      local names = table.concat(servers, server_name_sep, start_index, _end_index)
+      names = header_empty and header .. names or space:rep(#header) .. names
+      table.insert(lines, space)
+      table.insert(lines, names)
+      length = #header
+      start_index = i + 1
+      _end_index = 0
+    end
+  end
+
+  return lines, hi_scope
+end
```

Comment:
I might be missing something, but wouldn't enabling the `'wrap'` window option solve this in an easier way?

Reply:
yup use wrap it's easy.  But it doesn't look so pretty

Reply:
I beg to differ :D. Left is new, right is `master`. This is a 50% vertical split on a MacBook Air.

<img width="1363" alt="Screenshot 2022-08-24 at 16 52 27" src="https://user-images.githubusercontent.com/6705160/186451229-3c7b8503-09d0-400e-8b82-024cd5ad0b5c.png">

To me it seems like quite a lot of formatting code for a subjectively better UI - just my 2 cents though!

Reply:
FWIW, the indentation of wrapped lines can be adjusted with `breakindent` and `breakindentopt`.

Reply:
see the pic 2 .I still think there should be have space between the line

Reply:
so how ?

![image](https://user-images.githubusercontent.com/41671631/186536559-c341445f-4409-40c5-8b9d-eb54aff68822.png)